File: ac044ccd-3d72-4455-8c1c-58b7a9637580-f03a43a3-5f6a-4811-aa2c-6c609cec2ffb-main/springapp/src/main/java/com/examly/springapp/service/WiFiSchemeService.java
package com.examly.springapp.service;

import java.util.List;

import com.examly.springapp.exceptions.DuplicateWiFiSchemeException;
import com.examly.springapp.exceptions.WiFiSchemeDeletionException;
import com.examly.springapp.exceptions.WiFiSchemeException;
import com.examly.springapp.model.WiFiScheme;

public interface WiFiSchemeService {
    WiFiScheme addWiFiScheme(WiFiScheme wifiScheme) throws DuplicateWiFiSchemeException, WiFiSchemeException;

    WiFiScheme getWiFiSchemeById(Long wifiSchemeId) throws WiFiSchemeException;

    List<WiFiScheme> getAllWiFiSchemes();

    WiFiScheme updateWiFiScheme(Long wifiSchemeId, WiFiScheme wifiScheme);

    WiFiScheme deleteWiFiScheme(Long wifiSchemeId) throws WiFiSchemeDeletionException;

}

File: ac044ccd-3d72-4455-8c1c-58b7a9637580-f03a43a3-5f6a-4811-aa2c-6c609cec2ffb-main/angularapp/src/app/services/wifi-scheme.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http'
import { WifiScheme } from '../models/wifi-scheme.model';
import { Observable } from 'rxjs';


@Injectable({
  providedIn: 'root'
})
export class WifiSchemeService {


  apiUrl:string="https://8080-fadbadfaddcabbbeddcbfaafaaacccecffb.premiumproject.examly.io/api/wifiScheme";
  
  constructor(private http:HttpClient) {}

    public getAllWiFiSchemes():Observable<WifiScheme[]>{
    return this.http.get<WifiScheme[]>(this.apiUrl);
  }

  public getWiFiSchemeById(wifiSchemeId:number):Observable<WifiScheme>{
    return this.http.get<WifiScheme>(this.apiUrl+"/"+wifiSchemeId);
  }

  public addWiFiScheme(wifiScheme:WifiScheme):Observable<WifiScheme>{
    return this.http.post<WifiScheme>(this.apiUrl,wifiScheme);
  }

  public updateWiFiScheme(wifiSchemeId:number, wifiScheme:WifiScheme):Observable<WifiScheme>{
    return this.http.put<WifiScheme>(this.apiUrl+"/"+wifiSchemeId,wifiScheme);
  }

  public deleteWiFiScheme(wifiSchemeId:number): Observable<any>{
     return this.http.delete<WifiScheme>(this.apiUrl+"/"+wifiSchemeId);
  }


}
